File: Seccion 12 - Punteros en C++/Seccion_12_Dart/08_ejercicio_08.dart

import 'dart:io';

String palabraUsuario;

void pedirCadena() {

  stdout.write('Digite una palabra o frase: ');
  palabraUsuario = stdin.readLineSync().toUpperCase();

}

void contarVocal(String palabra, String vocal) {

  int cont = 0;

  for (int i = 0; i < palabra.length; i++) {
    if (palabra[i] == vocal) {
      cont++;
    }
  }

  stdout.writeln('Número de veces que aparece la vocal $vocal: $cont');
}

void main() {

  pedirCadena();

  print('');
  contarVocal(palabraUsuario, 'A');
  contarVocal(palabraUsuario, 'E');
  contarVocal(palabraUsuario, 'I');
  contarVocal(palabraUsuario, 'O');
  contarVocal(palabraUsuario, 'U');
  
}

File: Seccion 12 - Punteros en C++/Seccion_12_Python/08_ejercicio_08.py

palabra_ususario: str = ''

def pedir_cadena():
    global palabra_ususario
    
    palabra_ususario = input('Digite una palabra o frase: ')
    palabra_ususario = palabra_ususario.upper()

def contar_vocal(palabra: str, vocal: str):
    
    cont: int = 0
    
    for letra in palabra:
        if letra == vocal:
            cont += 1
        
    print('Número de veces que aparece la vocal {}: {}'.format(vocal, cont))

def main():
    
    pedir_cadena()
    
    print()
    contar_vocal(palabra_ususario, 'A')
    contar_vocal(palabra_ususario, 'E')
    contar_vocal(palabra_ususario, 'I')
    contar_vocal(palabra_ususario, 'O')
    contar_vocal(palabra_ususario, 'U')
    
if __name__ == '__main__':
    main()